sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarStorageStrategy.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import io.opentelemetry.sdk.metrics.view.Aggregation;
+
+/** A strategy for storing exemplars. */
+@FunctionalInterface
```

Comment:
Confirm it's worth adding this with the tradeoff of not being able to add any default methods in the future. This doesn't seem like such a "functional" class

Reply:
Not sure.  There's a future PR to redo the `Aggregation` API so this method can be fully implemented.   From my perspective this should MOSTLY be a pattern match against aggregation type, optionally pulling out some aggregation config.

Reply:
As far as I know, you can add `default` methods to `@FunctionalInterface` SMIs without breaking anything.

Reply:
But, I agree that this probably isn't adding a lot of value at the moment, since the JVM treats all SMIs as `@FunctionalInterface`s by default.

Reply:
if we wanted this to use built-in types, we could make it a `Function<Aggregation, ExemplarReservoir>`. That might end up being more confusing, though.

Reply:
I expect the interface to (eventually) have more arguments (or a diffferent argument than just the raw "Aggregation", that's in a follow-up CL where we wire in the HistogramExemplarReservoir + DefaultExemplarReservoir, so going to avoid `Function` as we might need more than one parameter.

Took a crack at fixing LMKWYT.